src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -489,6 +491,23 @@ private static boolean equalsItemMeta(@Nonnull ItemMeta itemMeta, @Nonnull ItemM
             return ((PotionMeta) itemMeta).getBasePotionData().equals(((PotionMeta) sfitemMeta).getBasePotionData());
         }
 
+        // Fixes #4020: Book and Head Player Comparison
+
+        if (itemMeta instanceof BookMeta itemBookMeta && sfitemMeta instanceof BookMeta sfItemBookMeta) {
+            if(itemBookMeta.getAuthor() != sfItemBookMeta.getAuthor())
+                return false;
+            if(itemBookMeta.getTitle() != sfItemBookMeta.getTitle())
+                return false;
+            if(itemBookMeta.getPageCount() != sfItemBookMeta.getPageCount())
+                return false;
+        }
+
+        if (itemMeta instanceof SkullMeta itemHeadMeta && sfitemMeta instanceof SkullMeta sfItemHeadMeta) {
+            if(itemHeadMeta.getOwningPlayer() != sfItemHeadMeta.getOwningPlayer())
+                return false;
+        }
```

Comment:
Other than the codestyle this looks good, only thing I think is wrong is the if statements are missing brackets and some spaces. 

Other than that functionally its probably okay but you can inline the returns to be like
`return itemBookMeta.getAuthor().equals(sfItemBookMeta.getAuthor() && etc etc`
or
return `itemHeadMeta.getOwningPlayer().equals(sfItemHeadMeta.getOwningPlayer());`
I'm not 100% sure that the player check is a good idea since it's nullable iirc, so you might need to change that, I'd have to check to be completely sure